src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/DatesViewModelToChangeOfEmployerRequestMapper.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.CommitmentsV2.Shared.Interfaces;
+using SFA.DAS.ProviderCommitments.Web.Models.Apprentice;
+
+namespace SFA.DAS.ProviderCommitments.Web.Mappers.Apprentice
+{
+    public class DatesViewModelToChangeOfEmployerRequestMapper : IMapper<DatesViewModel, ChangeOfEmployerRequest>
+    {
+        private readonly ILogger<DatesViewModelToChangeOfEmployerRequestMapper> _logger;
+
+        public DatesViewModelToChangeOfEmployerRequestMapper(ILogger<DatesViewModelToChangeOfEmployerRequestMapper> logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task<ChangeOfEmployerRequest> Map(DatesViewModel source)
+        {
+            try
+            {
+                return await Task.FromResult(new ChangeOfEmployerRequest
+                {
+                    ProviderId = source.ProviderId,
+                    ApprenticeshipHashedId = source.ApprenticeshipHashedId,
+                    EmployerAccountLegalEntityPublicHashedId = source.EmployerAccountLegalEntityPublicHashedId,
+                    StartDate = source.StartDate.MonthYear,
+                    Price = source.Price.Value
+                });
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Error Mapping {nameof(DatesViewModel)} to {nameof(ChangeOfEmployerRequest)}", e);
```

Comment:
Do we need to catch and log? If an exception is thrown, the call stack will give us the name of the mapper won't it? And since the mapper only has one job - to map from x to y - the log message is self-evident?

Reply:
I can see what you mean, but the pattern of logging errors as close to the failure as possible makes tracing easier. But will remove if you want me to.